requests.md

```diff
@@ -1,26 +1,70 @@
 # Requests
 
+## Path resolution
+
+Chopper handles paths passed to HTTP verb annotations' `path` parameter based on the path's content.
+
+If the `path` value is a relative path, it will be concatenated to the URL composed of the `baseUrl` of the `ChopperClient` and the `baseUrl` of the enclosing service class (provided as a parameter of the `@ChopperApi` annotation).
+
+Here are a few examples of the described behavior:
+
+ * `ChopperClient` base URL: https://example.com/
+    Path: profile
+    Result: https://example.com/profile
+
+* `ChopperClient` base URL: https://example.com/
+  Service base URL: profile
+  Path: /image
+  Result: https://example.com/profile/image
+
+* `ChopperClient` base URL: https://example.com/
+  Service base URL: profile
+  Path: image
+  Result: https://example.com/profile/image
+
+> Chopper detects and handles missing slash (`/`) characters on URL segment borders, but *does not* handle duplicate slashes.
+
+If the service's `baseUrl` concatenated with the request's `path` results in a full URL, the `ChopperClient`'s `baseUrl` is ignored.
+
+`ChopperClient` base URL: https://example.com/
+Service base URL: https://api.github.com/
+Path: user
+Result: https://api.github.com/user
+
+A `path` containing a full URL replaces the base URLs of both the `ChopperClient` and the service class entirely for a request.
+
+* `ChopperClient` base URL: https://example.com/
+  Path: https://api.github.com/user
+  Result: https://api.github.com/user
+* `ChopperClient` base URL: https://example.com/
+  Service base URL: profile
+  Path: https://api.github.com/user
+  Result: https://api.github.com/user
+
+
 ## Path parameters
 
-Use `{}`to specify the name and the position of your parameter directly in the url
+Path parameters can be defined in the URL with replacement blocks. A replacement block is an alphanumeric substring of the path surrounded by `{` and `}`. In the following example `{id}` is a replacement block.
 
 ```dart
-@Get(path: '/{id}')
+@Get(path: "/{id}")
```

Comment:
Hmm, I did not notice this earlier, but why the double quotes? I saw many examples with single and using them in my working projects.

Reply:
Well, Dart actually allows both single and double quoted String literals, so both should work as expected (like in JavaScript).

However, the Dart linter actually gives a strange "silent" warning for single quoted String literals in Android Studio:

![image](https://user-images.githubusercontent.com/5860632/103488698-35432180-4e0f-11eb-931e-50f1d5933812.png)

I've investigated this warning further and I found 2 conflicting linter rules:

* https://dart-lang.github.io/linter/lints/prefer_single_quotes.html (Effective from Dart 2.0.0)
* https://dart-lang.github.io/linter/lints/prefer_double_quotes.html (Effective from Dart 2.3.2-dev.0.0)

The latter being in effect with the current stable Flutter+Dart SDK.

![image](https://user-images.githubusercontent.com/5860632/103488928-e8604a80-4e10-11eb-9d40-25c1568f9e1f.png)

I've decided to follow the linter, that's all. 😄

Reply:
Cool, thanks for investigation!